h3/src/proto/headers.rs

```diff
@@ -21,10 +21,14 @@ pub struct Header {
 
 #[allow(clippy::len_without_is_empty)]
 impl Header {
-    pub fn request(method: Method, uri: Uri, fields: HeaderMap) -> Self {
-        Self {
-            pseudo: Pseudo::request(method, uri),
-            fields,
+    pub fn request(method: Method, uri: Uri, fields: HeaderMap) -> Result<Self, Error> {
+        match (uri.authority(), fields.get("host")) {
```

Comment:
Is the `host` header supposed to be a thing in HTTP/3? I think it's replaced by `:authority`, right?

Reply:
The `:authority` field is the preffered way of checking if the peer has the sufficient authority regarding its presented certificate. But `Host` can be used for that purpose:

In [ 4.1.1.1. Pseudo-Header Fields](https://quicwg.org/base-drafts/draft-ietf-quic-http.html#name-pseudo-header-fields):

> If the ":scheme" pseudo-header field identifies a scheme that has a mandatory authority component (including "http" and "https"), the request MUST contain either an ":authority" pseudo-header field or a "Host" header field. If these fields are present, they MUST NOT be empty. If both fields are present, they MUST contain the same value. If the scheme does not have a mandatory authority component and none is provided in the request target, the request MUST NOT contain the ":authority" pseudo-header or "Host" header fields.

Do you want me to add this paragraph as a comment ?

Reply:
Oh cool, the spec is explicit about it. In the HTTP2 spec, it basically says host is dead unless coming from an HTTP/1 proxy. It's fine whichever you want to do here.